core/Controller/ProfileApiController.php

```diff
@@ -80,4 +91,51 @@ public function setVisibility(string $targetUserId, string $paramId, string $vis
 
 		return new DataResponse();
 	}
+
+	/**
+	 * Get profile fields for another user
+	 *
+	 * @param string $targetUserId ID of the user
+	 * @return DataResponse<Http::STATUS_OK, CoreProfileData, array{}>|DataResponse<Http::STATUS_BAD_REQUEST|Http::STATUS_NOT_FOUND, null, array{}>
+	 *
+	 * 200: Profile data returned successfully
+	 * 400: Profile is disabled
+	 * 404: Account not found or disabled
+	 */
+	#[NoAdminRequired]
+	#[ApiRoute(verb: 'GET', url: '/{targetUserId}', root: '/profile')]
+	#[BruteForceProtection(action: 'user')]
+	#[UserRateLimit(limit: 30, period: 120)]
+	public function getProfileFields(string $targetUserId): DataResponse {
+		$targetUser = $this->userManager->get($targetUserId);
+		if (!$targetUser instanceof IUser) {
+			$response = new DataResponse(null, Http::STATUS_NOT_FOUND);
+			$response->throttle();
+			return $response;
+		}
+		if (!$targetUser->isEnabled()) {
+			return new DataResponse(null, Http::STATUS_NOT_FOUND);
+		}
+
+		if (!$this->profileManager->isProfileEnabled($targetUser)) {
+			return new DataResponse(null, Http::STATUS_BAD_REQUEST);
+		}
+
+		$requestingUser = $this->userSession->getUser();
+		if ($targetUser !== $requestingUser) {
+			if (!$this->shareManager->currentUserCanEnumerateTargetUser($requestingUser, $targetUser)) {
+				return new DataResponse(null, Http::STATUS_NOT_FOUND);
+			}
+		}
+
+		$profileFields = $this->profileManager->getProfileFields($targetUser, $requestingUser);
+
+		// Extend the profile information with timezone of the user
+		$timezoneStringTarget = $this->config->getUserValue($targetUser->getUID(), 'core', 'timezone') ?: $this->config->getSystemValueString('default_timezone', 'UTC');
+		$timezoneTarget = new \DateTimeZone($timezoneStringTarget);
```

Comment:
Timezone string from user config could theoretically be BS.

```suggestion
try {
		$timezoneTarget = new \DateTimeZone($timezoneStringTarget);
} catch (\DateInvalidTimeZoneException) {
		// whatever you want to do to handle the exception - return default?
}
```

Reply:
> This method throws [DateInvalidTimeZoneException](https://www.php.net/manual/en/class.dateinvalidtimezoneexception.php) if the timezone supplied is not recognised as a valid timezone. Prior to PHP 8.3, this was an [Exception](https://www.php.net/manual/en/class.exception.php) instead.